### ISSUE TITLE
[java] UnusedImports false positive for method parameter type in @see Javadoc

### ISSUE THREAD

**Issue by rillig-tk** [pmd/pmd#1555]
**Affects PMD Version:**
* master as of 2019-01-04
* 6.*
* 5.*

**Rule:**
UnusedImports

**Description:**
When a derived class refers to a type in a `@see #method(ImportedType)` Javadoc tag, the `ImportedType` is flagged as being unused.

**Code Sample demonstrating the issue:**

```java
public class Base {
    void method(java.io.File file) {}
}
```

```java
import java.io.File;

public class Derived extends Base {
    /** @see #method(File) */
    void main() {}
}
```

**Running PMD through:** doesn't matter

Suggested fix: In `UnusedImportsRule.java`, change `SEE_PATTERN`:

```diff
- "@see\\s+(\\p{Alpha}\\w*)(?:#\\w*(?:\\(([\\w\\s,]*)\\))?)?"
+ "@see\\s+(\\p{Alpha}?+\\w*)(?:#\\w*(?:\\(([\\w\\s,]*)\\))?)?"
```

This makes the type name optional. The regular expression should now match:

* `Typename#method(Typename)` (already worked before)
* `#method(Typename)` (new)

**Comment by Shubham-2k17** [comment]
How to create test case for **multiple public class** as shown in given sample? All test cases in **UnusedImports.xml** have only one public class in each test-code. Any documentation or reference will be helpful. The pull request #1625 is missing test case for given sample.

**Comment by jsotuyod** [comment]
@Shubham-2k17 you can include base classes in the test code. That way they will be compiled and included.

For instance, we declare a custom `Error` class on:

https://github.com/pmd/pmd/blob/master/pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/design/donotextendjavalangerror/Error.java

And then the test-case extends it on:

https://github.com/pmd/pmd/blob/dd66f9c5a2b96889780a5995f1919293c91a624d/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/design/xml/DoNotExtendJavaLangError.xml#L37-L48

Just make sure to include the proper import!
